src/java.base/share/classes/java/lang/Class.java

```diff
@@ -241,6 +241,7 @@ private Class(ClassLoader loader, Class<?> arrayComponentType) {
         // prevents future JIT optimizations from assuming this final field is null.
         classLoader = loader;
         componentType = arrayComponentType;
+        modifiers = 0;
```

Comment:
The comment above about assigning a parameter to the field to prevent the JIT from assuming an incorrect default also should apply to the new `modifiers` field.  I think the constructor, which is never called, should also pass in a `dummyModifiers` value rather than using 0 directly